### DISCUSSION TITLE
與蜘蛛展開對話 美科學家將蜘蛛活動頻率轉化為樂曲

### DISCUSSION THREAD

**Discussion by 239216** [1101-GS4719-C/Python-Community-GS4719-C#54] [General]
布德勒表示，蜘蛛在蜘蛛網上活動時，會震動蜘蛛網，藉此與外界溝通；這些細微的聲音富含許多資訊，如交響樂般豐富。「我們錄下這些細微聲響，運用人工智慧得出這些聲響的頻率，並觀測蜘蛛在何種頻率下會做出何種動作，就能大略學會蜘蛛的語言。」

布德勒教授和團隊重新打造出3D蜘蛛網模型。他們精密觀察蜘蛛活動所產生的頻率，並使用科技將頻率轉化為樂音。
蜘蛛是很另類的生物，但藉由我們的科學實驗，大眾得以進入蜘蛛的世界」。

https://news.ltn.com.tw/news/novelty/breakingnews/3500463

**Comment by chia-kai-chang** [comment]
查了一下這是2019年的MIT新聞稿：https://news.mit.edu/2019/spider-web-music-inspiring-harmony-art-and-science-0108

對於自由時報記者的翻譯有些疑問耶：**計算頻率要用FFT，不需要用AI**

物理系、電機系的同學有沒有要補充的?

**Reply by enoch-lightning** [reply]
我雖然不是物理系或電機系，我也想嘗試補充。FFT為fast fourier transform的縮寫，中文為「快速傅利葉轉換」。在我們所處的世界，所有的訊號在時間上都是連續(continuous)的，也稱為類比(analog)訊號。若要分析這些訊號的頻率，便需使用電腦。電腦最基本的處理單位是0與1，也就是以時間「離散(discrete)」的方式處理資訊，這種訊號稱為數位(digital)訊號。要將類比訊號轉為數位訊號，會經過取樣(sample)，以固定或變動的時間間隔取點。若數位訊號有一基礎週期(最小週期)，以此週期無限重複，便可使用「離散傅利葉級數(Discrete-Time Fourier Series)(DTFS)」分析，將得出此數位訊號為由特定某幾個頻率的訊號組成，即在頻率上為離散。但，現實生活中不會收到時間無限長的類比訊號，進而取樣成時間無限長的數位訊號。因此，這時的數位訊號為非週期性函數，便需使用「離散傅利葉轉換(Discrete-Time Fourier Transform)(DTFT)」分析頻率。而FFT，即為以演算法的方式提昇離散傅利葉轉換的運算速度。我雖不是很了解，但將目前整理的理解提供於此，可進一步討論。

1.何謂連續(continuous)？
答案：當一函數的定義域(domain)與對應域(codomain)均為實數的子集合(subset)，在某一點的極限值(limit)=此點的函數值，即稱函數在此點連續。

**Comment by Jie2222** [comment]
[FFT](https://blog.csdn.net/shenziheng1/article/details/52891807)
>**節錄文章內容** :  如果採樣頻率Fs为1024Hz，採樣點數為1024點，則可以分辨到1Hz。1024Hz的採樣率採樣1024點，剛好是1秒，也就是說，採樣1秒時間的信號並做FFT，則结果可以分析到1Hz，如果採樣2秒时间的信號並做FFT，相應的採樣點也為原来2倍，則结果可以分析到0.5Hz。如果要提高频率分辨力，則必须增加採樣點數，也即延長採樣時間，所以频率分辨率和採樣時間是倒數關係，就是说，要想分辨出頻率間隔越小的頻率（頻率分辨率越高），採樣時間越長越好。

雖然看不是很懂，但也許對於回答問題會有幫助

**Comment by chia-kai-chang** [comment]
關於FFT，youtube上面有一些動畫解說，做的很生動，有興趣可以看一下：
1. https://youtu.be/r18Gi8lSkfM
2. https://youtu.be/spUNpyF58BY
